### PR TITLE
feat: optionally target any buffer

### PR DESCRIPTION
**Pull request by molleweide**

### CODE REVIEW

**Comment by molleweide** [comment]
I have to read up on whether or not one has to modify any of `autoload` or the `plugins` dir also now since `run` accepts four params.

##edit 

Nah, I think this is the only required change, since the Ex command definition has nothing to do with how the `run` func works.

File: lua/architext/cmd.lua [hunk 1]
```diff
@@ -134,9 +134,11 @@ local function parse_argument(parser, text)
   return query, changes
 end
 
-function M.run(text, start_row, end_row)
+function M.run(text, start_row, end_row, buf)
 
-  local buf = a.nvim_get_current_buf()
+  if buf == nil then
+    buf = a.nvim_get_current_buf()
+  end
```
Comment (vigoux) [inline lua/architext/cmd.lua]: Far more idiomatic:
```suggestion
    buf = buf or a.nvim_get_current_buf()
```

**Review by vigoux — APPROVED** [review]
Looks pretty good, I'll check a bit more later and merge.

File: lua/architext/cmd.lua (hunk 1 above)
Comment (molleweide) [reply lua/architext/cmd.lua]: yes haha!!!!!

**Review by vigoux — APPROVED** [review]

**Pull request merged**